src/legacy/core_plugins/kibana/public/discover/np_ready/angular/directives/field_name/field_name.tsx

```diff
@@ -64,7 +64,14 @@ export function FieldName({ field, fieldName, fieldType, useShortDots, fieldIcon
         />
       </EuiFlexItem>
       <EuiFlexItem className="eui-textTruncate">
-        <span className="dscFieldName__displayName eui-textTruncate">{displayName}</span>
+        <EuiToolTip
+          position="top"
+          content={displayName}
+          delay="long"
+          anchorClassName="eui-textTruncate"
```

Comment:
dear masters of design, dear @elastic/kibana-design 
I wonder why there are 2 nested `eui-textTruncate` needed to truncate the text? Doesn't work with one.
thx!

Reply:
Getting text to truncate is tricky. Why? Because browsers suck and they haven't figured a better way yet. But sometime every wrapper of the child will need a `overflow: hidden` in order for the truncation to work. So easiest solution, use the class for each block wrapper.

Reply:
@cchaos thx for the explanation! 👍